Backend/controllers/userController.js

```diff
@@ -0,0 +1,193 @@
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+const userService = require("../services/userService");
+const authService = require("../services/authService");
+const hashPassword = require("../utils/hashPassword");
+const dotenv = require("dotenv");
+dotenv.config();
+
+const salt = process.env.SALT;
+
+const validateSignup = (req) => {
+  const { userName, email, password } = req.body;
+  const errors = [];
+
+  if (!userName || userName.trim() === "") {
+    errors.push("Username is required");
+  }
+
+  if (!email || email.trim() === "") {
+    errors.push("Email is required");
+  } else if (!isValidEmail(email)) {
+    errors.push("Email is invalid");
+  }
+
+  if (!password || password.trim() === "") {
+    errors.push("Password is required");
+  } else if (password.length < 6) {
+    errors.push("Password should be at least 6 characters long");
+  }
+
+  return errors;
+};
+
+const validateLogin = (req) => {
+  const { email, password } = req.body;
+  const errors = [];
+
+  if (!email || email.trim() === "") {
+    errors.push("Email is required");
+  } else if (!isValidEmail(email)) {
+    errors.push("Email is invalid");
+  }
+
+  if (!password || password.trim() === "") {
+    errors.push("Password is required");
+  }
+
+  return errors;
+};
+
+const validateForgotPassword = (req) => {
+  const { email } = req.body;
```

Comment:
use proper naming conventions for functions.

why are you validating email inside validateForgotPassword function?